src/etc/dl-snapshot.py

```diff
@@ -6,22 +6,40 @@
 import tarfile
 import shutil
 import contextlib
+import re
 
+datere = re.compile('^\d{4}-\d{2}-\d{2}')
+cksumre = re.compile('^  ([^ ]+) ([^$]+)$')
+
+current = None
+snaps = {}
 with open('src/snapshots.txt') as f:
-    lines = f.readlines()
-
-date = lines[0]
-bitrig64 = lines[1]
-linux32 = lines[2]
-linux64 = lines[3]
-mac32 = lines[4]
-mac64 = lines[5]
-win32 = lines[6]
-win64 = lines[7]
+    for line in iter(f):
+        line = line.rstrip()
+        m = datere.match(line)
+        if m:
+            current = m.group()
+            snaps[current] = {}
+            continue
+
+        m = cksumre.match(line)
+        if m:
+            snaps[current][m.group(1)] = m.group(2)
+            continue
+
+        # This script currently doesn't look at older snapshots, so there is
+        # no need to look past the first section.
+        break
+
+date = snaps.keys()[0]
```

Comment:
This has blown up Python 3 support, because `keys()` returns a set-like object rather than a list as it did in Python 2.